osf_tests/metrics/reporters/test_institutional_users_reporter.py

```diff
@@ -0,0 +1,257 @@
+from __future__ import annotations
+import dataclasses
+import datetime
+import unittest
+
+from django.test import TestCase
+
+from api_tests.utils import create_test_file
+from osf import models as osfdb
+from osf.metrics.reports import InstitutionalUserReport
+from osf.metrics.reporters import InstitutionalUsersReporter
+from osf.metrics.utils import YearMonth
+from osf_tests.factories import (
+    InstitutionFactory,
+    PreprintFactory,
+    ProjectFactory,
+    RegistrationFactory,
+    UserFactory,
+    EmbargoFactory,
+)
+
+
+def _can_affiliate_preprints() -> bool:
+    # HACK: preprints affiliation project still in-progress
+    return hasattr(osfdb.Preprint, 'affiliated_institutions')
+
+
+def _patch_now(fakenow: datetime.datetime):
+    return unittest.mock.patch('django.utils.timezone.now', return_value=fakenow)
+
+
+class TestInstiUsersReporter(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls._yearmonth = YearMonth(2012, 7)
+        cls._now = datetime.datetime(
+            cls._yearmonth.year,
+            cls._yearmonth.month,
+            13,  # just some day in the month
+            tzinfo=datetime.UTC,
+        )
+        with _patch_now(cls._now):
+            cls._institution = InstitutionFactory()
+            cls._user_setup_with_nothing = _InstiUserSetup(0, 0, 0, 0, 0, cls._institution, cls._now)
+            cls._user_setup_with_ones = _InstiUserSetup(1, 1, 1, 1, 1, cls._institution, cls._now)
+            cls._user_setup_with_stuff = _InstiUserSetup(
+                2, 3, 5, 3, 2, cls._institution, cls._now,
```

Comment:
nit: I feel like it's more readable with keyword argument for these fiddly test numbers